### PR TITLE
Joeychrys/issue11

### PR DESCRIPTION
**Pull request by joeychrys**
Hey I created a rating component. Cheers!

### CODE REVIEW

File: src/components/rating/RatingContext.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export function useRatingContext(): RatingContext {
+  const context = useContext(RatingContext);
+
+  if (!context) {
+    throw new Error('useRatingContext should be used within the RatingContext provider!');
+  }
+
+  return context;
+}
```
Comment (joeychrys) [inline src/components/rating/RatingContext.tsx:18]: I created a context so there isn't a need to pass the size to every `Rating.Star`.

Comment (joeychrys) [inline src/components/rating/RatingContext.tsx:18]: I added a context so there isn't a need to pass the `size` to every `Rating.Star`.

**Comment by zoltanszogyenyi** [comment]
Hey @joeychrys @bacali95,

Just wanted to say you're doing an awesome job!

The core team (including myself) from the main library is now working on a new release and after that we'll hop in to help with the docs of the React library.

**Comment by joeychrys** [comment]
> 

Awesome! Let me know if you need any help on the docs.

**Comment by joeychrys** [comment]
> Hey @joeychrys @bacali95,
> 
> Just wanted to say you're doing an awesome job!
> 
> The core team (including myself) from the main library is now working on a new release and after that we'll hop in to help with the docs of the React library.

Awesome! Let me know if you need any help with the docs.

File: src/components/rating/Rating.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { RatingAdvanced } from './RatingAdvanced';
+import { RatingContext } from './RatingContext';
+import { RatingStar } from './RatingStar';
+
+type Size = 'sm' | 'md' | 'lg';
```
Comment (bacali95) [inline src/components/rating/Rating.tsx:7]: You can export this type to be used from `RatingContext`

File: src/components/rating/RatingContext.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { ComponentProps } from 'react';
+import { createContext, useContext } from 'react';
```
Comment (bacali95) [inline src/components/rating/RatingContext.tsx:2]: Please merge these imports

Comment (bacali95) [inline src/components/rating/RatingContext.tsx:2]: Please merge these imports

File: src/components/rating/RatingContext.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+export type RatingContext = ComponentProps<'svg'> & {
+  size?: string;
```
Comment (bacali95) [inline src/components/rating/RatingContext.tsx:5]: You can use the type from the `Rating` component

Comment (bacali95) [inline src/components/rating/RatingContext.tsx:5]: You can use the type from the `Rating` component

File: src/components/rating/RatingContext.tsx (hunk 1 above)
Comment (bacali95) [reply src/components/rating/RatingContext.tsx:18]: Nice :grin:

File: src/components/rating/RatingStar.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  sm: 'w-5 h-5',
+  md: 'w-7 h-7',
+  lg: 'w-10 h-10',
+};
+
+export const RatingStar: FC<RatingStarProps> = ({ filled = true }) => {
+  const { size } = useRatingContext();
+
+  return (
+    <svg
```
Comment (bacali95) [inline src/components/rating/RatingStar.tsx:19]: You can use `HiIcon` from the `react-icons` library it is the same one as this, I checked

Comment (bacali95) [inline src/components/rating/RatingStar.tsx:19]: You can use `HiIcon` from the `react-icons/hi` it is the same one as this, I checked

**Review by bacali95 — CHANGES_REQUESTED** [review]
I just added support to React 18 in the `main` branch which requires some changes to make it work, can you please rebase and apply what I did to these components?

**Comment by joeychrys** [comment]
> I just added support to React 18 in the `main` branch which requires some changes to make it work, can you please rebase and apply what I did to these components?

I sure can!

File: src/components/rating/Rating.tsx [hunk 6]
```diff
@@ -1,11 +1,9 @@
...
 import { RatingAdvanced } from './RatingAdvanced';
-import { RatingContext } from './RatingContext';
+import { RatingContext, Size } from './RatingContext';
 import { RatingStar } from './RatingStar';
 
-type Size = 'sm' | 'md' | 'lg';
-
 export type RatingComponentProps = {
   className?: string;
   size?: Size;
```
Comment (joeychrys) [inline src/components/rating/Rating.tsx:9]: Should I use `PropsWithChildren` here?

Comment (joeychrys) [inline src/components/rating/Rating.tsx:9]: Should I add `PropsWithChildren` here?

File: src/components/rating/RatingContext.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { ComponentProps, createContext, useContext } from 'react';
+
+export type Size = 'sm' | 'md' | 'lg';
+
+export type RatingContext = ComponentProps<'svg'> & {
```
Comment (bacali95) [inline src/components/rating/RatingContext.tsx:5]: Why use the `ComponentProps<'svg'>` props? I don't see it used in the child components

**Review by bacali95 — CHANGES_REQUESTED** [review]

File: src/components/rating/Rating.tsx (hunk 6 above)
Comment (bacali95) [reply src/components/rating/Rating.tsx:9]: I think yes you should do that.

File: src/components/rating/RatingContext.tsx (hunk 7 above)
Comment (joeychrys) [reply src/components/rating/RatingContext.tsx:5]: oh yeah I forgot to remove that.

**Review by bacali95 — APPROVED** [review]
Nice job @joeychrys, I will merge this one.

**Pull request merged**